### PR TITLE
Add AksTestRunner and replace TestController

### PR DESCRIPTION
**Pull request by ziyuezh576**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                .WithProjectSubfolderForTests ("ScenarioTests")
+                .WithCommonPsScripts (new[]
+                {
+                    @"Common.ps1",
+                    @"../AzureRM.Resources.ps1"
+                })
+                .WithNewRmModules (helper => new[]
+                {
+                    helper.RMProfileModule,
+                    helper.GetRMModulePath("AzureRM.Aks.psd1")
```
Comment (vidai-msft) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:35]: Update from AzureRM.Aks.psd1 to Az.Aks.psd1

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                        {"Microsoft.Azure.Management.Resources.ResourceManagementClient", "2017-05-10"},
+                        {"Microsoft.Azure.Management.ResourceManager.ResourceManagementClient", "2017-05-10"}
+                    },
+                    resourceProviders: new Dictionary<string, string>
+                    {
+                        {"Microsoft.Resources", null},
+                        {"Microsoft.Features", null},
+                        {"Microsoft.Authorization", null}
+                    }
+                ).WithMockContextAction(() =>
```
Comment (vidai-msft) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:49]: Remove the invocation of WithMockContextAction()

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                        if (currentEnvironment.ConnectionString.KeyValuePairs.ContainsKey("ServicePrincipalSecret"))
+                        {
+                            spnSecret = currentEnvironment.ConnectionString.KeyValuePairs["ServicePrincipalSecret"];
+                        }
+                        AzureSession.Instance.DataStore.WriteFile(Path.Combine(home, ".ssh", "id_rsa.pub"), File.ReadAllText(dir + "/Fixtures/id_rsa.pub"));
+                        var jsonOutput = @"{""" + subscription + @""":{ ""service_principal"":""" + spn + @""",""client_secret"":"""+ spnSecret + @"""}}";
+                        AzureSession.Instance.DataStore.WriteFile(Path.Combine(home, ".azure", "acsServicePrincipal.json"), jsonOutput);
+                    }
+                }
+                ).WithManagementClients(
```
Comment (vidai-msft) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:83]: Remove the invocation of WithManagementClients()

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                    GetContainerServiceClient,
+                    GetGraphManagementClient,
+                    GetAuthorizationManagementClient,
+                    GetInternalResourceManagementClient
+                )
+                .Build();
+        }
+
+        private static ContainerServiceClient GetContainerServiceClient(MockContext context)
```
Comment (vidai-msft) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:92]: This method definition is not necessary. Please remove it.

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            return context.GetServiceClient<ContainerServiceClient>();
+        }
+        private MicrosoftGraphClient GetGraphManagementClient(MockContext context)
```
Comment (vidai-msft) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:96]: This method definition is not necessary. Please remove it.

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            return context.GetServiceClient<MicrosoftGraphClient>();
+        }
+        private static AuthorizationManagementClient GetAuthorizationManagementClient(MockContext context)
```
Comment (vidai-msft) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:100]: This method definition is not necessary. Please remove it.

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            return context.GetServiceClient<AuthorizationManagementClient>();
+        }
+        private static ResourceManagementClient GetInternalResourceManagementClient(MockContext context)
```
Comment (vidai-msft) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:104]: This method definition is not necessary. Please remove it.

**Review by vidai-msft — CHANGES_REQUESTED** [review]
Please check the review comments and change the code accordingly.

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs (hunk 2 above)
Comment (vidai-msft) [reply src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:49]: WithMockContextAction() still not removed

**Review by vidai-msft — CHANGES_REQUESTED** [review]
WithMockContextAction() still not removed.

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs (hunk 2 above)
Comment (ziyuezh576) [reply src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs]: I deleted my previous comment by mistake. I mean, I can't pass any ScenarioTests if I remove it, Is there no problem when you run?

Comment (vidai-msft) [reply src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:49]: Looks like ssh related code is necessary.

**Review by vidai-msft — APPROVED** [review]
Looks good !

File: src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs [hunk 8]
```diff
@@ -0,0 +1,87 @@
+﻿using System;
```
Comment (isra-fel) [inline src/Aks/Aks.Test/ScenarioTests/AksTestRunner.cs:1]: License header

**Review by isra-fel — CHANGES_REQUESTED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**